torch/optim/optimizer.py

```diff
@@ -394,6 +394,7 @@ def zero_grad(self, set_to_none: bool = False):
             per_device_and_dtype_grads = defaultdict(lambda: defaultdict(list))
         with torch.autograd.profiler.record_function(self._zero_grad_profile_name):
             for group in self.param_groups:
+                tmp_group = defaultdict(list)
```

Comment:
No need for this, you already have the per_device_and_dtype_grads

Reply:
fixed in new commit https://github.com/pytorch/pytorch/pull/91392/commits/33bda4b6c96d74724d52719920b19dbe7e8bff92